### DISCUSSION TITLE
Error while trying to export to CSV

### DISCUSSION THREAD

**Discussion by ni3ls** [code4recovery/12-step-meeting-list#477] [Q&A] [answered]
Does anyone else see an error when trying to export the data in Version 3.11.2?

Error:

2021/07/28 18:17:07 [error] 32568#32568: *45109 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "location_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_url_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_phone_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "location_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_url_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_phone_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-
2021/07/28 18:17:07 [error] 32568#32568: *45109 FastCGI sent in stderr: "otes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_url_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_phone_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_url_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_phone_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "location_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_url_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "conference_phone_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "location_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/ajax.php on line 238PHP message: PHP Warning:  Undefined array key "group_notes" in /var/www/html/aacostadelsol.com/wp-content/plugins/12-step-meeting-list/includes/aja
2021/07/28 18:17:07 [error] 32568#32568: *45109 FastCGI sent in stderr: "-content/plugins/12-step-meeting-list/includes/ajax.php on line 235" while reading response header from upstream, client: 79.147.236.107, server: aacostadelsol.com, request: "GET /wp-admin/admin-ajax.php?action=csv HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.0-fpm.sock:", host: "aacostadelsol.com", referrer: "https://aacostadelsol.com/wp-admin/edit.php?post_type=tsml_meeting&page=import"

Maybe I missed a database migration along the way?

Thanks for any hints

**Comment by tech2serve** [comment, marked as answer]
Hey @ni3ls--

To answer your first question, I do not get any errors when doing an export of meeting data in v3.11.2.

Regarding your second question, I'm not sure what is going on. Did you recently upgrade TSML? If so, what was the previous version? And does this issue correlate to that upgrade?

If you're not able to export meetings, I think your options may be a bit limited. Hopefully you have a backup of your meetings that is reasonably current.

Not sure how many meetings you have (I couldn't load your site) but you might try going into them to edit and update and see if you get any errors. 

It might just be quicker and to delete and reinstall the plugin (you can try activate/deactivate?), then import meeting data from your backup.

**Reply by ni3ls** [reply]
Thank you so much - I am in progress of re-establishing everything so I will close this matter again - It is very clear that there is an inconsistency in the database vs. the export "ajax.php" file - to be honest not even worth bothering too much about :-)

Thanks all for your great work with this plugin!

/N

**Comment by ni3ls** [comment]
UPDATE:

Tested a new install of the plugin, the error persist!
BUT!

**I think it is related to me using PHP version 8.0**
It seems like the file: "ajax.php" is not yet upgraded to this version.

Does anyone else use PHP version 8.0 and can confirm that this is the issue?
Looks to me like the way arrays are used is changed in this upgrade :-(

Thanks in advance
